From c34e25bc9519eee11334f30857f66a8483bd32f7 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Thu, 01 Jul 2010 18:11:40 +0000
Subject: Drop up front call to pager_update

It's going to get called at realize time anyway,
and calling it early can actually be counter-productive,
since it checks state that may not be properly
initialized yet.

https://bugzilla.gnome.org/show_bug.cgi?id=472773
(cherry picked from commit 819b938de2b1c1805e2dc6b69866418a361c5244)
---
diff --git a/applets/wncklet/workspace-switcher.c b/applets/wncklet/workspace-switcher.c
index 02d18a8..584eec2 100644
--- a/applets/wncklet/workspace-switcher.c
+++ b/applets/wncklet/workspace-switcher.c
@@ -541,11 +541,6 @@ workspace_switcher_applet_fill (PanelApplet *applet)
 	gtk_container_add (GTK_CONTAINER (pager->applet), pager->pager);
 	gtk_widget_show (pager->pager);
 
-	/* we need to call this after the pager has been added to the
-	 * container, else it might not work in some cases in multiscreen
-	 * environments. See wnck_pager_set_orientation() doc */
-	pager_update (pager);
-
 	g_signal_connect (G_OBJECT (pager->applet),
 			  "realize",
 			  G_CALLBACK (applet_realized),
--
cgit v0.8.3.1

